### PR TITLE
ROX-22724: increase reprocessor concurrency

### PR DESCRIPTION
**Pull request by stehessel**
## Description

Increase the reprocessor concurrency limit fro 5 to 10 and expose it as an environment variable. This is done for two reasons:
* to counter balance the increased registry timeouts, which increase the risk of clogging the reprocessor.
* 5 concurrent images is a low value, which likely can be increased further. The reasoning behind the low value was an attempt to restrict the memory usage in Central. However with the recent migration to an external database and other optimizations, this no longer seems to be a blocker.

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

### Here I tell how I validated my change

TODO(replace-me)  
Use this space to explain **how you validated** that **your change functions exactly how you expect it**.
Feel free to attach JSON snippets, curl commands, screenshots, etc. Apply a simple benchmark: would the information you
provided convince any reviewer or any external reader that you did enough to validate your change.

It is acceptable to assume trust and keep this section light, e.g. as a bullet-point list.

It is acceptable to skip testing in cases when CI is sufficient, or it's a markdown or code comment change only.  
It is also acceptable to skip testing for changes that are too taxing to test before merging. In such case you are
responsible for the change after it gets merged which includes reverting, fixing, etc. Make sure you validate the change
ASAP after it gets merged or explain in PR when the validation will be performed.  
Explain here why you skipped testing in case you did so.

Have you created automated tests for your change? Explain here which validation activities you did manually and why so.

### Reminder for reviewers

In addition to reviewing code here, reviewers **must** also review testing and request further testing in case the
performed one does not seem sufficient. As a reviewer, you must not approve the change until you understand the
performed testing and you are satisfied with it.

### CODE REVIEW

**Comment by stehessel** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/10432?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#10432** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/10432?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#10431** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/10431?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#10430** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/10430?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @stehessel and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: pkg/env/reprocessor.go [hunk 1]
```diff
@@ -0,0 +1,6 @@
+package env
+
+var (
+	// ReprocessorSemaphoreLimit is the maximum number of reprocessed images.
```
Comment (dhaus67) [inline pkg/env/reprocessor.go:4]:
```suggestion
	// ReprocessorSemaphoreLimit is the maximum number of images being reprocessed in parallel.
```

File: pkg/env/reprocessor.go [hunk 2]
```diff
@@ -0,0 +1,6 @@
...
+	ReprocessorSemaphoreLimit = RegisterIntegerSetting("ROX_REPROCESSOR_SEMAPHORE_LIMIT", 10)
```
Comment (dhaus67) [inline pkg/env/reprocessor.go:5]: Hmm, I wonder if we should be more "aggressive" and bump it to the same amount as https://github.com/stackrox/stackrox/blob/master/pkg/env/max_parallel_scans.go; but this probably does warrant looking at the scale tests more to see the effect of it, so feel free to ignore.

**Review by dhaus67 — COMMENTED** [review]
Would be great to run scale tests on these to see the effect of things, even if it is to just get a feel of it

File: pkg/env/reprocessor.go (hunk 2 above)
Comment (stehessel) [reply pkg/env/reprocessor.go:5]: Let's increase it gradually. I also think that we can likely go to something like 30 images based on the memory profile of our big cloud tenants (barely any reprocessor spike currently).

**Review by janisz — APPROVED** [review]

**Pull request closed without merging**